core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -438,15 +439,24 @@ angular.module('oppia').directive('questionsList', [
                   ).then(function() {
                     QuestionsListService.resetPageNumber();
                     QuestionsListService.getQuestionSummariesAsync(
-                      ctrl.selectedSkillId, true, true
-                    );
+                      ctrl.selectedSkillId, true, true);
                     AlertsService.addSuccessMessage('Deleted Question');
+                    ctrl.removeArrayElement(
+                      ctrl.questionIdsDeletedArray, questionId);
                   });
                 }
               });
             }
           };
 
+          ctrl.removeArrayElement = function(array, n) {
```

Comment:
Let's make this a private method `_removeDeletedQuestion`.

Reply:
Also, we don't really need to pass in `array` right? Since we have `ctrl.deletedQuestionIds`.

Maybe change n -> `questionId`